File: src/2/2.py
# Python: Demonstrating dynamic typing and function scope (closures)
def outer():
    x = 10
    def inner():
        # Python uses lexical scoping. 'nonlocal' allows modifying the variable in the enclosing scope.
        nonlocal x
        x += 1
        return x
    return inner

closure_function = outer()
print("Python closure call 1:", closure_function())  # 11
print("Python closure call 2:", closure_function())  # 12

File: src/2/2.js
// JavaScript: Demonstrating lexical scoping and closures
function outer() {
    let x = 10;
    function inner() {
        // JavaScript functions create closures capturing variables from the outer scope.
        x += 1;
        return x;
    }
    return inner;
}

const closureFunction = outer();
console.log("JavaScript closure call 1:", closureFunction());  // 11
console.log("JavaScript closure call 2:", closureFunction());  // 12


// Ouput
// JavaScript closure call 1: 11
// JavaScript closure call 2: 12
